donrec.php

```diff
@@ -280,15 +280,17 @@ function donrec_civicrm_mailjet_transactional_bounce($bounce_message) {
  * Will inject the FastActivity tab
  */
 function donrec_civicrm_tabset($tabsetName, &$tabs, $context) {
-  if ($tabsetName == 'civicrm/contact/view' && !empty($context['contact_id'])) {
-    if (CRM_Core_Permission::check('view and copy receipts') || CRM_Core_Permission::check('create and withdraw receipts')) {
+  if ($tabsetName == 'civicrm/contact/view') {
+    // If contact_id is not provided, this is being called from the ContactLayout extension to retrieve tab info
+    if (!$context['contact_id'] || CRM_Core_Permission::check('view and copy receipts') || CRM_Core_Permission::check('create and withdraw receipts')) {
```

Comment:
Are you sure this is right? Before, it would only add the tab if the contact ID was known (i.e. ``$context['contact_id']`` wasn't empty) - now it would add it with no ``contact_id`` as long as the permissions are granted.

Reply:
I think there's always a contact ID when the tabset is `civicrm/contact/view`, either with explicit `$context` or implicit through *ContactLayoutEditor*'s implementation, at least that's how I get @colemanw's comment. Would there be any scenario the contact tabs wouldn't be displayed for a contact?

But, fair point being, checking for `$context['contact_id]` not being truthy seems indeed wrong. Also, `$context['contact_id]` is required for the tab's URL 2 lines down, so it will have to be there, right?

Reply:
> But, fair point being, checking for `$context['contact_id]` not being truthy seems indeed wrong. Also, `$context['contact_id]` is required for the tab's URL 2 lines down, so it will have to be there, right?

Yes, that's exactly my concern

Reply:
And add 142a18faee45f796c71bfdf78ca8d528ccb92be8 for avoiding `Undefined index` notices ...